include/deal.II/lac/read_write_vector.h

```diff
@@ -283,7 +283,7 @@ namespace LinearAlgebra
      */
     void import(const distributed::Vector<Number> &vec,
                 VectorOperation::values operation,
-                std::shared_ptr<const CommunicationPatternBase> communication_pattern =
+                const std::shared_ptr<const CommunicationPatternBase> &communication_pattern =
```

Comment:
Personally, I don't like this. Why would we want to pass a  `shared_ptr` by reference? There is also no point in making the `shared_ptr` constant, what matters is what the `shared_ptr` points to.

Reply:
This way we avoid increasing and decreasing the counter though.